RecoEgamma/Examples/plugins/SimplePi0DiscAnalyzer.cc

```diff
@@ -11,39 +11,75 @@
      <Notes on implementation>
 */
 //
-// Original Author:  Aristotelis Kyriakis
+// Original Author:  Aristotelis Kyriakis NCSR "Demokritos" Athens
+//                    D Maletic, "Vinca" Belgrade
 //         Created:  May 26 13:22:06 CEST 2009
 //
 //
 
-// user include files
-#include "RecoEgamma/Examples/plugins/SimplePi0DiscAnalyzer.h"
-#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/EcalDetId/interface/EcalSubdetector.h"
+#include "DataFormats/EgammaCandidates/interface/Photon.h"
+#include "DataFormats/EgammaCandidates/interface/PhotonFwd.h"
+#include "DataFormats/EgammaCandidates/interface/PhotonPi0DiscriminatorAssociation.h"
 #include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/ESHandle.h"
 #include "FWCore/Framework/interface/Event.h"
-#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
-#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
-#include "DataFormats/EgammaCandidates/interface/GsfElectron.h"
-#include "DataFormats/EgammaCandidates/interface/GsfElectronFwd.h"
-#include "DataFormats/EgammaReco/interface/BasicClusterFwd.h"
-#include "DataFormats/EgammaReco/interface/SuperClusterFwd.h"
```

Comment:
I don't think so, because if a function can return a `SuperClusterRef` it means that `SuperClusterFwd.h` must already be included in the header file where said function is defined, no?

Reply:
So, if you define a `SuperClusterRef` as r`eco::Photon::superCluster()`, then you must have the header of `Photon` included here, which necessarily has a `SuperClusterFwd.h` included in its turn... Ok, it makes sense!